hack/bin/integration-logs-relevant-bits.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+USAGE="
+Filter out garbage from logs (but keep colors and highlight problems).
+
+(Adapted from logfilter.sh by Stefan Matting)
+
+To use it pipe log output from integration-test.sh into this tool.
+
+Usage: logs | $0 [options]
+"
+
+red_color="\x1b\[;1m\x1b\[31m"
+problem_markers="Failures:|FAILED|ExitFailure""\
+|\^+""\
+|FAIL""\
+|tests failed""\
+|Test suite failure""\
+|""$red_color""error:""\
+|""$red_color""\
+|Test suite .+ failed"
+
+exit_usage() {
+    echo "$USAGE"
+    exit 1
+}
+
+# remove debug/info logs
+# often this is just noise like connection to cassandra.
+excludeLogEntries() {
+    grep -v '^{".*Info"' |
+        grep -v '^{".*Debug"'
+}
+
+cleanup() {
+    # replace backspaces with newlines
+    # remove "Progress" lines
+    # Remove blank lines
+    # add newline between interleaved test name and log output lines
+    sed 's/\x08\+/\n/g' |
+        sed '/^Progress [0-9]\+/d' |
+        sed '/^\s\+$/d' |
+        sed 's/:\s\+{/:\n{/g'
+}
+
+grepper() {
+    # print 10 lines before/after for context
+    rg "$problem_markers" --color=always -A 10 -B 10
```

Comment:
I think this hopes that the red colour is reset during the next 10 lines and no other colour is set at that point, which might not be true and mess up rest of the output. I think we should explicitly print a colour reset code after this `rg` to ensure that doesn't happen.

Reply:
this is kind of the end of the ouput for the integration tests. I've not observde any problems with colour so far. But happy to add something. Would a `echo -e "\033[0m"` line after the `rg` line do the trick?